src/picongpu/include/plugins/BinEnergyParticles.hpp

```diff
@@ -163,10 +163,11 @@ __global__ void kernelBinEnergyParticles(ParticlesBox<FRAME, simDim> pb,
                  * but without if*/
                 binNumber = binNumber * (int) (binNumber < maxBin) +
                     maxBin * (int) (binNumber >= maxBin);
-                /* same as
-                 * binNumber>0?binNumber:0
-                 * but without if*/
-                binNumber = (int) (binNumber > 0) * binNumber;
+
+                /* Warning:
+                 * use `binNumber>0?binNumber:0` instead of `int(binNumber > 0) * binNumber`
+                 * second one does not work due to unknown reason */
```

Comment:
I agree on that, even if this solves the problem. 
We need to **understand** what is happening here.